### ISSUE TITLE
Feature Request: SSL configuration support for external database connections

### ISSUE THREAD

**Issue by adjohns** [jhpyle/docassemble#345]
## Feature Request: SSL configuration support for external database connections
Please add support to the configuration file to allow the database to use Secure Sockets Layer (SSL) connections. This may also need to support taking in a file path or S3/Blob location for a trusted certificate for the ssl connection to use. It looks like the [psycopg]( https://www.psycopg.org/docs/module.html) module will support the sslmode parameters and cert paths out of the box.

When deploying DocAssemble on a single machine or cluster not having SSL enforcement is not as big of an issue since the traffic is not leaving the machine. However, when deploying the web server to one machine and the DB separately the traffic from the front end may have to traverse a public or private network. This risks exposure to man-in-the-middle type attacks or compromise to the data.

**Example of potential config file layout:**
db:
&nbsp;prefix: postgresql+psycopg2://
&nbsp;name: docassemble
&nbsp;user: docassemble
&nbsp;password: abc123
&nbsp;host: database.hostname.domain.com
&nbsp;port: 5432
&nbsp;table prefix: mycustomprefix
&nbsp;backup: False
&nbsp;_**ssl mode**: require | allow | prefer | disable | verify-ca | verify-full_
&nbsp;_**db cert**: blob://youraccountname/yourcontainername/dbcerts_

`conn = psycopg2.connect(database=db_name, user=db_user, password=db_password, host=db_host, port=db_port, sslmode=ssl_mode, sslrootcert=~/.postgresql/root.crt)`

Thank you for considering this.

**Comment by jhpyle** [comment]
This is a great suggestion. Thank you!

**Comment by jhpyle** [comment]
I added this feature in 1.2.24.  See https://docassemble.org/docs/config.html#db

You will need to do a system upgrade first.  While no docker container is running, I would recommend changing your config.yml in your Azure Blob Storage (or wherever you keep it) to add `ssl mode` etc.  Add the certificate files to the `certs` folder in your Blob Storage.  Then do `docker start`.  When the server starts, it will copy the certificate files from the cloud to the appropriate place on the local machine.

I got it to work in testing by generating all three certificate files and installing them on the PostgreSQL server, then including the files in my certs folder and specifying their filenames in my config.yml.  I also set `ssl mode: require`.  The file paths are passed to SQLAlchemy as "connection arguments."

Let me know if you have any questions.

**Comment by adjohns** [comment]
@jhpyle - Amazing, thank you for adding this in! I just pulled the latest code for 1.2.24 and rebuilt the image, tested and everything works smoothly and as expected.

Using [Azure database for PostgreSQL with SSL configured](https://docs.microsoft.com/azure/postgresql/concepts-ssl-connection-security) and the DA configuration file stored in a Storage Account. I copied over the single root.crt file to the /certs path in the Blob and specified the configurations of: **DBSSLMODE=verify-full**, **DBSSLROOTCERT= root.crt** in addition to all the usual DB configurations (name, database, host, etc).

Really appreciate your dedication and continuous improvements on the project!
